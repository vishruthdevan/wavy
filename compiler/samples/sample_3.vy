let samples = [2, 4, -1, 5];
let volumeBoost = fn(sample) { return sample * 10; };

let adjustedSample1 = volumeBoost(samples[0]);
let adjustedSample2 = volumeBoost(samples[1]);
puts(adjustedSample1);
puts(adjustedSample2);