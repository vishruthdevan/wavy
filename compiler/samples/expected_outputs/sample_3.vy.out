0000 OpConstant 0
0003 OpConstant 1
0006 OpConstant 2
0009 OpMinus
0010 OpConstant 3
0013 OpArray 4
0016 OpSetGlobal 0
0019 OpClosure 5 0
0023 OpSetGlobal 1
0026 OpGetGlobal 1
0029 OpGetGlobal 0
0032 OpConstant 6
0035 OpIndex
0036 OpCall 1
0038 OpSetGlobal 2
0041 OpGetGlobal 1
0044 OpGetGlobal 0
0047 OpConstant 7
0050 OpIndex
0051 OpCall 1
0053 OpSetGlobal 3
0056 OpGetBuiltin 1
0058 OpGetGlobal 2
0061 OpCall 1
0063 OpPop
0064 OpGetBuiltin 1
0066 OpGetGlobal 3
0069 OpCall 1
0071 OpPop
